pandas/tests/util/test_assert_index_equal.py

```diff
@@ -280,3 +280,11 @@ def test_assert_index_equal_object_ints_order_false():
     idx1 = Index([1, 3], dtype="object")
     idx2 = Index([3, 1], dtype="object")
     tm.assert_index_equal(idx1, idx2, check_order=False)
+
+
+def test_assert_ea_index_equal_non_matching_na():
+    # GH#48608
+    idx1 = Index([1, 2], dtype="Int64")
+    idx2 = Index([1, NA], dtype="Int64")
+    with pytest.raises(AssertionError, match="50.0 %"):
+        tm.assert_index_equal(idx1, idx2)
```

Comment:
Does this also work if `check_exact` or `check_categorical` are `False`?

Reply:
categorical=False did not work, but fixed it